src/main/resources/com/cloudbees/plugins/credentials/impl/CertificateCredentialsImpl/UploadedKeyStoreSource/configUpload.js

```diff
@@ -0,0 +1,54 @@
+// multiple objects named "password" in the form =>
+// extend findNearBy to allow selecting by id
+const oldNearBy = findNearBy;
+findNearBy = function(el, name) {
+    return name.charAt(0) == '#' ? document.querySelector(name.split('/')[0]) : oldNearBy(el, name);
+}
+
+Behaviour.specify(".certificate-file-upload", 'certificate-file-upload', -99, function(uploadedCertFileInput) {
+  // Adding a onChange method on the file input to retrieve the value of the file content in a variable
+  var fileId = uploadedCertFileInput.id;
+  var _onchange = uploadedCertFileInput.onchange;
+  if (typeof _onchange === "function") {
+    uploadedCertFileInput.onchange = function() { fileOnChange(this); _onchange.call(this); }
+  } else {
+    uploadedCertFileInput.onchange = fileOnChange.bind(uploadedCertFileInput);
+  }
+  const base64field = uploadedCertFileInput.closest('.radioBlock-container').querySelector('[name="certificateBase64"]');
+  function fileOnChange() {
+    try { // inspired by https://stackoverflow.com/a/754398
+      var uploadedCertFileInputFile = uploadedCertFileInput.files[0];
+      var reader = new FileReader();
+      reader.onload = function (evt) {
+        base64field.value = btoa(evt.target.result);
+        var uploadedKeystore = document.getElementById(fileId + "-textbox");
+        uploadedKeystore.onchange(uploadedKeystore);
+      }
+      reader.onerror = function (evt) {
+        if (window.console !== null) {
+          console.warn("Error during loading uploadedCertFile content", evt);
+        }
+        uploadedCertFile[fileId] = '';
+      }
+
+      reader.readAsBinaryString(uploadedCertFileInputFile);
+    }
+    catch(e){
+      if (window.console !== null) {
+        console.warn("Unable to retrieve uploadedCertFile content");
+      }
+    }
+  }
+
+  // workaround for JENKINS-19124
+  // without this script, the password changes will be not trigger the check on the uploadedKeystore
```

Comment:
The code changed significantly but the reasoning is still the same: I couldn't find a way how to make `password` textfield trigger validation of the cert field, since there are multiple fields named `password` in the form and they are on higher level than the cert upload field.

Reply:
` @RealativePath(..)` is the way.  There is a bug https://issues.jenkins.io/browse/JENKINS-65616 that causes this to work for some validation triggers, but not others.

Reply:
The relative path can't be just `password` because the field is on a higher level. But the relative path `../password` matches the wrong password.